manager/controllers/package_detail.go

```diff
@@ -0,0 +1,118 @@
+package controllers
+
+import (
+	"app/base/database"
+	"app/base/models"
+	"app/base/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/jinzhu/gorm"
+	"net/http"
+)
+
+func singlePackageLatestQuery(pkgName string) *gorm.DB {
+	return database.Db.Table("package pkg").Select("*").Joins("JOIN package_name pn ON pkg.name_id = pn.id").
+		Joins("JOIN advisory_metadata adv ON pkg.advisory_id = adv.id").
+		Where("pn.name = ?", pkgName).
+		Order("adv.public_date DESC").Limit(1)
+}
+
+func singlePackageEvraQuery(pkgName string, evra string) *gorm.DB {
+	return database.Db.Table("package").
+		Joins("JOIN package_name ON package.name_id = package_name.id").
+		Where("package_name.name = ?", pkgName).
+		Where("package.evra = ?", evra)
+}
+
+func getAdvisoryByID(advID int) (models.AdvisoryMetadata, error) {
+	var advisory models.AdvisoryMetadata
+	err := database.Db.Where("id = ?", advID).First(&advisory).Error
+	return advisory, err
+}
+
+type PackageDetailResponse struct {
+	Data SinglePackage `json:"data"`
+	Name string        `json:"name"`
+}
+
+type SinglePackage struct {
+	Package  models.Package          `json:"package"`
+	Advisory models.AdvisoryMetadata `json:"advisory"`
+}
+
+func packageLatestHandler(c *gin.Context, packageName string) {
+	query := singlePackageLatestQuery(packageName)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespError(c, err, "database error")
+		return
+	}
+
+	var advisory models.AdvisoryMetadata
+	err = database.Db.Where("id = ?", packageInfo.AdvisoryID).First(&advisory).Error
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	c.JSON(200, PackageDetailResponse{
+		Data: SinglePackage{
+			Package:  packageInfo,
+			Advisory: advisory,
+		},
+		Name: packageName,
+	})
+}
+
+func packageEvraHandler(c *gin.Context, nevra *utils.Nevra) {
+	packageName := nevra.Name
+	evra := nevra.EVRAString()
+
+	query := singlePackageEvraQuery(packageName, evra)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
```

Comment:
What if it's not found?

Reply:
Then 500 is returned with message that specified package was not found.

Reply:
5xx status codes should be returned only on error cases what's not on "Not found". It should return 404 status in that case. Again see the AdvisoryDetailHandler and implement is similarly.

Reply:
Noticed, returning 404.